pkgs/development/haskell-modules/lib.nix

```diff
@@ -196,6 +196,16 @@ rec {
   appendPatch = drv: x: appendPatches drv [x];
   appendPatches = drv: xs: overrideCabal drv (drv: { patches = (drv.patches or []) ++ xs; });
 
+  /* Set a specific build target instead of compiling all targets in the package.
+   * For example, imagine we have a .cabal file with a library, and 2 executables "dev" and "server".
+   * We can build only "server" and not wait on the compilation of "dev" by using setBuildTarget as follows:
+   *
+   *   setBuildTarget (callCabal2nix "thePackageName" thePackageSrc {}) "server"
+   *
+   */
+  setBuildTargets = drv: xs: overrideCabal drv (drv: { buildTarget = lib.concatStringsSep " " xs; });
```

Comment:
Replacing `buildTarget` with a list parameter called `buildTargets` would be nice, but I'm not sure to what extent `buildTarget` may be considered public API?

Reply:
Ah, yeah, that is a good idea.  Although I'd argue that it shouldn't block this PR getting merged in as it currently as.

As for whether it would be considered a public API, I'm not sure.  I've wondered about that as well.  We have the functions in `pkgs/development/haskell-modules/lib.nix` that allow us to set the arguments in `generic-builder.nix`, but we don't have a function in `pkgs/development/haskell-modules/lib.nix` for _every_ argument in `generic-builder.nix`, so sometimes you have to set them directly.  It is also more convenient to set the arguments directly sometimes rather than going through the functions in pkgs/development/haskell-modules/lib.nix.

Reply:
As this hasn't been working properly until now, we probably can change stuff about it without to much problems, right?

Reply:
Ah, I see what you're saying.  Since `buildTarget` hasn't actually been working correctly, then we don't really need to consider it part of the public API.

I think that makes sense, and it sounds like it makes sense to make that change.

Although I don't think it should necessarily stop this PR from being merged in.

Reply:
It has been working correctly for libraries